client/src/components/Grid/GridList.vue

```diff
@@ -0,0 +1,292 @@
+                    :filter-class="config.filtering"
+                    :filter-text.sync="filterText"
+                    :loading="loading"
+                    :show-advanced.sync="showAdvanced"
+                    @on-backend-filter="onSearch" />
+                <hr v-if="showAdvanced" />
+            </div>
+            <div v-if="!showAdvanced" class="py-3">
+                <BButton
+                    v-for="(action, actionIndex) in config.actions"
+                    :key="actionIndex"
+                    class="m-1"
+                    size="sm"
+                    variant="primary"
+                    :data-description="`grid action ${action.title.toLowerCase()}`"
+                    @click="action.handler(router)">
+                    <Icon :icon="action.icon" class="mr-1" />
+                    <span v-localize>{{ action.title }}</span>
+                </BButton>
+            </div>
+        </div>
+        <LoadingSpan v-if="loading" />
+        <BAlert v-else-if="!isAvailable" variant="info" show>
+            <span v-if="searchTerm" v-localize>
+                No entries found for: <b>{{ searchTerm }}</b
+                >.
+            </span>
+            <span v-else v-localize> No entries found. </span>
+        </BAlert>
+        <table v-else class="grid-table">
+            <thead>
+                <th
+                    v-for="(fieldEntry, fieldIndex) in config.fields"
+                    :key="fieldIndex"
+                    class="text-nowrap px-2"
+                    :data-description="`grid header ${fieldIndex}`">
+                    <span v-if="config.sortKeys.includes(fieldEntry.key)">
+                        <BLink @click="onSort(fieldEntry.key)">
+                            <span>{{ fieldEntry.title || fieldEntry.key }}</span>
+                            <span v-if="sortBy === fieldEntry.key">
+                                <FontAwesomeIcon v-if="sortDesc" icon="caret-up" data-description="grid sort asc" />
+                                <FontAwesomeIcon v-else icon="caret-down" data-description="grid sort desc" />
+                            </span>
+                        </BLink>
+                    </span>
+                    <span v-else>{{ fieldEntry.title || fieldEntry.key }}</span>
+                </th>
+            </thead>
+            <tr v-for="(rowData, rowIndex) in gridData" :key="rowIndex" :class="{ 'grid-dark-row': rowIndex % 2 }">
+                <td
+                    v-for="(fieldEntry, fieldIndex) in config.fields"
+                    :key="fieldIndex"
+                    class="px-2 py-3"
+                    :style="{ width: `${fieldEntry.width}%` }"
+                    :data-description="`grid cell ${rowIndex}-${fieldIndex}`">
+                    <GridOperations
+                        v-if="fieldEntry.type == 'operations'"
+                        :operations="fieldEntry.operations"
```

Comment:
There is still a type error here. Either relax the parameter on `GridOperations`, or make the type more concrete, or use a runtime type-assertion, to resolve it.

Reply:
<img width="571" alt="image" src="https://github.com/galaxyproject/galaxy/assets/2105447/9cd9f5e5-90b0-4e1e-a66d-da5670277f03">

Hmm, this is how this code section looks to me. Maybe there is an additional plugin I'd have to install for the editor.

Reply:
It sounds like your IDE may not be using the projects tsconfig. That would be my first guess.

Reply:
Another Option is that the prop types are not resolving properly. You can check this by hovering over some typed props in a consuming/parent component, and checking if the types displayed are correct. In case they are not: are you on the newest volar and typescript tolar versions?

Reply:
Added a PR to resolve this typescript warning: #17065.